src/Language/PureScript/Types.hs

```diff
@@ -77,7 +90,7 @@ data Type a
   -- | Explicit kind application
   | KindApp a (Type a) (Type a)
   -- | Forall quantifier
-  | ForAll a Text (Maybe (Type a)) (Type a) (Maybe SkolemScope)
+  | ForAll a Text (Maybe (Type a)) (Type a) (Maybe SkolemScope) TypeVarVisibility
```

Comment:
IIRC, the values stored in these constructors are usually stored in the same order of appearance. Nit, but shouldn't the `TypeVarVisbility` arg appear before the `Text` arg?

Reply:
I think that's ideal, yep

Reply:
On a second look, I think it's more apt to have it at the end as the Aeson instances for `Type` makes use of tuples for the ForAll. I'm not sure if this affects anything though

Reply:
Ah... like backwards compatibility on Pursuit might be affected?

Reply:
Yeah, to be exact, this change in particular:
https://github.com/purescript/purescript/blob/a227bdd5a71616dffe09c83c26cfa6c097efa7de/src/Language/PureScript/Types.hs#L369-L381

I do think that this is the correct behaviour though—it retains backward compatibility by parsing the contents in a way that the most "general" case is at the end. Tests also pass with this implementation.

Reply:
Why do we need to keep using Tuples for this particular encoding? Couldn't we define the representation as an object and try decoding that representation first? And if it fails, we fallback to the original tuple encoding? I believe that would still maintain backwards compatibility, but it would also allow `Forall` to be extended in the future without running into this order-dependent issue again.
```haskell
let 
  objMbKind = A.withObject "Forall" $ \o -> do
   visibility <- o .: "visibility"
   identifier <- o .: "identifier"
   k <- o .:? "kind"
   ty <- o .: "type"
   skolemScope <- o .: "skolemScope"
   pure $ Forall a visibility identifier k ty skolemScope
   
   withoutMbKind = do
     (b, c, d) <- contents
     ForAll a TypeVarInvisible b Nothing <$> go c <*> pure d
  
   withMbKind = do   
     (b, c, d, e) <- contents  
     ForAll a TypeVarInvisible b <$> (Just <$> go c) <*> go d <*> pure e
  
objWithMbKind <|> withMbKind <|> withoutMbKind

Reply:
That's a good idea. I think it also makes sense to remove this backwards compatibility in the (far) future.